### PR TITLE
More MyST functionality

### PR DESCRIPTION
**Pull request by fwkoch**
This PR adds:

- roles
  - subscript
  - superscript
  - underline
  - inline math
  - abbreviations
  - content references (for figures, equations, headings)
- directives
  - admonitions
  - figure containers (for images, tables)
  - math blocks
  - tables
  - extending of commonmark image and code blocks
- footnotes
- comments
- block breaks

Additional improvements (or at least changes 🙂 ) to the schema:
- stricter validation of properties (making `additionalProperties: false` work) - this builds confidence that the examples are correct, without typos, etc.
- deduplication of `required` fields
- eliminating `oneOf` in favor of `anyOf` everywhere - ran into some tedious, incorrect errors due to `oneOf`.

It also adds a handful of additional commonmark examples.

Still todo:
- panels
- dropdowns
- definitions
- epigraphs
- glossaries
- indexes
- tabbed content
- citations
- ...

### CODE REVIEW

File: schema/commonmark/commonmark.paragraphs.yml [hunk 1]
```diff
@@ -56,3 +60,5 @@ cases:
       **strong**, _emphasis_, `literal text`, \*escaped symbols\*
     html: |-
       <p><strong>strong</strong>, <em>emphasis</em>, <code>literal text</code>, *escaped symbols*</p>
+    latex: |-
+      \textbf{strong}, \textit{emphasis}, \textttf{literal text}, *escaped symbols*
```
Comment (rowanc1) [inline schema/commonmark/commonmark.paragraphs.yml:64]:
```suggestion
      \textbf{strong}, \textit{emphasis}, \texttt{literal text}, *escaped symbols*
```

File: schema/directives/directives.admonitions.simple.yml [hunk 2]
```diff
@@ -22,6 +22,11 @@ cases:
         <p class="admonition-title">Attention</p>
         <p>Admonition attention</p>
       </aside>
+    latex: |-
+      \begin{callout}
```
Comment (rowanc1) [inline schema/directives/directives.admonitions.simple.yml:26]: This should be mdframed likely.

File: schema/directives/directives.image.yml [hunk 3]
```diff
@@ -39,6 +39,8 @@ cases:
       ```
     html: |-
       <img src="fun-fish.png" alt="fishy" class="align-center bg-primary mb-1" width="200px">
+    latex: |-
+      \includegraphics[width=200px]{fun-fish.png}
```
Comment (rowanc1) [inline schema/directives/directives.image.yml:43]: The measurements is going to be a bit tough in latex. Doesn't do px.

**Review by rowanc1 — COMMENTED** [review]
Minor latex comments!

**Comment by fwkoch** [comment]
This PR resolves #6 #7 #9 as well as @rowanc1 's minor latex comments.

I will address #10 and #11 in a separate PR

**Pull request merged**